org.knime.scijava.commands/src/org/knime/scijava/commands/module/DefaultNodeModuleService.java

```diff
@@ -0,0 +1,173 @@
+import org.scijava.Identifiable;
+import org.scijava.module.ModuleInfo;
+import org.scijava.module.ModuleItem;
+import org.scijava.module.ModuleService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.service.AbstractService;
+
+/**
+ * Default implementation of {@link NodeModuleService}.
+ *
+ * @author Christian Dietz, University of Konstanz
+ */
+@Plugin(type = NodeModuleService.class)
+public class DefaultNodeModuleService extends AbstractService
+        implements NodeModuleService {
+
+    @Parameter
+    private SettingsModelTypeService ts;
+
+    @Parameter
+    private ConverterCacheService cs;
+
+    @Parameter
+    private ModuleService ms;
+
+    @Override
+    public NodeModule createNodeModule(final ModuleInfo info,
+            final Map<String, SettingsModel> models, final DataTableSpec spec) {
+
+        final Map<Integer, ModuleItem<?>> inputMapping = new HashMap<>();
+        final Map<String, Object> params = new HashMap<String, Object>();
+
+        final Map<ModuleItem<?>, DataType> outputMapping = getOutputMapping(
+                info, models);
+        for (final ModuleItem<?> item : info.inputs()) {
+            if (MultiOutputListener.class.isAssignableFrom(item.getType())) {
+                continue;
+            }
+
+            final String name = item.getName();
+
+            final SettingsModelType<?, ?> type = ts
+                    .getSettingsModelTypeFor(item);
+
+            // FIXME make this extensible? special handling for special
+            // mappings or SettingsModels?
+            if (type instanceof SettingsModelColumnSelectionType) {
+                final SettingsModelColumnSelection settingsModel = (SettingsModelColumnSelection) models
+                        .get(item.getName());
+                inputMapping.put(
+                        spec.findColumnIndex(settingsModel.getStringValue()),
+                        item);
+            } else {
+                @SuppressWarnings("unchecked")
+                final SettingsModelType<SettingsModel, ?> tmp = ((SettingsModelType<SettingsModel, ?>) type);
+                params.put(name, tmp.getValue(models.get(name)));
+            }
+        }
+
+        return new DefaultNodeModule(getContext(), info, params, inputMapping,
+                outputMapping);
+
+    }
+
+    @Override
+    public DataTableSpec createOutSpec(final ModuleInfo info,
+            final Map<String, SettingsModel> models,
+            final DataTableSpec inSpec) {
+
+        final Map<ModuleItem<?>, DataType> outputMapping = getOutputMapping(
+                info, models);
+
+        final UniqueNameGenerator nameGen;
+        if (inSpec != null) {
+            nameGen = new UniqueNameGenerator(inSpec);
+        } else {
+            nameGen = new UniqueNameGenerator(new HashSet<>());
+        }
+
+        // TODO (What?)... What?
+        final List<DataColumnSpec> tableSpecs = new ArrayList<>();
```

Comment:
@gab1one @dietzc I don't think we'll get around ignoring that output for ScriptModules. After splitting the `DefaultNodeModule` into `ScriptNodeModule` and `CommandNodeModule`, that check will merely reside in the Script specific one anyway, though.